File: Codigos estudiantes por lenguaje/PY/Bryann Valderrama/Algoritmos de Ordenamiento/shellSort.py
'''Shell Sort

   - Algoritmo de Divide y Conquista.
   - Va comparando 2 numeros haciendo saltos.
   - Los saltos se reducen a la mitad en cada comparacion.
   - Complejidad Tiempo: 
        Peor: O(n^2) 
        Mejor: O (n log n)
'''

from sys import stdin, stdout
rl = stdin.readline
wr = stdout.write


def shellSort(arr=list, n=int):

    salto = n // 2
    while salto > 0:
        for i in range(salto, n):
            temp = arr[i]
            j = i
            while j >= salto and arr[j-salto] > temp:
                arr[j] = arr[j-salto]
                j -= salto
            arr[j] = temp
        salto //= 2


arr = list(map(int, rl().split()))
shellSort(arr, len(arr))
print(*arr)

File: Codigos estudiantes por lenguaje/Java/Sebastian_Acevedo/Ordenamiento/shellSort.java
package programming;
import java.util.Arrays;
import java.util.Scanner;
public class Main{
	
	
	static void shellSort(int arr[]) {
		int n = arr.length;
		for (int salto = n/2; salto>0; salto/=2) {
			for (int i = salto; i < n; i++) {
				int temp=arr[i];
				int j2;
				for (j2 = i; j2 >=salto&&arr[j2-salto]>temp; j2-=salto) {
					arr[j2]=arr[j2-salto];
				}
				arr[j2]=temp;
			}
		}
	}
	
	static void printArry(int arr[]) {
		for (int i = 0; i < arr.length; i++) {
			System.out.print(arr[i]+" ");
		}
		System.out.println();
	}
	
    public static void main(String[] args) {
    	Scanner sc = new Scanner(System.in);
    	int arr[]= {170, 45, 75, 90, 802, 24, 2, 66};
    	int n=arr.length;
        printArry(arr);
        shellSort(arr);
        printArry(arr);
    }
}
